File: tests/rules_lang/cc/fibonacci_main.cc
//! Fibonacci binary for printing the nth fibonacci number

#include <iostream>
#include <string>

#include "fibonacci.h"

bool is_number(const std::string& s) {
  std::string::const_iterator it = s.begin();
  while (it != s.end() && std::isdigit(*it)) ++it;
  return !s.empty() && it == s.end();
}

int main(int argc, char** argv) {
  if (argc != 2) {
    std::cerr << "Please provide a single integral input." << std::endl;
    return -1;
  }

  std::string argv1 = argv[1];

  if (!is_number(argv1)) {
    std::cerr << "The provided input is not an integer value." << std::endl;
    return -1;
  }

  std::uint64_t nth = std::stoull(argv1);

  std::cout << fibonacci::fibonacci(nth) << std::endl;
}

File: tests/rules_lang/rust/fibonacci_main.rs
//! Fibonacci binary for printing the nth fibonacci number

use std::env;
use std::process;

use fibonacci;

fn main() {
    if env::args().len() != 1 {
        eprintln!("Please provide a single integral input.");
        process::exit(1);
    }

    let argv1 = env::args().last().unwrap();

    let nth = match argv1.parse::<u64>() {
        Ok(val) => val,
        Err(e) => {
            eprintln!("{}", e);
            process::exit(1);
        }
    };

    println!("{}", fibonacci::fibonacci(nth));
}
